### PR TITLE
Fix local role deleting when role mapping is not added

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
Relates to wso2/product-is#7203

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java [hunk 1]
```diff
@@ -255,35 +255,40 @@ tobeDeleted claims (claims came from federated idp as null). If there is a match
                 diagnosticLog.info("Federated user: " + username + " is provisioned locally.");
             }
 
-            if (roles != null) {
-                // Update user with roles
-                List<String> currentRolesList = Arrays.asList(userStoreManager.getRoleListOfUser(username));
-                Collection<String> deletingRoles = retrieveRolesToBeDeleted(realm, currentRolesList, rolesToAdd);
+            boolean includeManuallyAddedLocalRoles = false;
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java:258]: false is redundant here

**Review by pulasthi7 — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java (hunk 1 above)
Comment (vihanga-liyanage) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java:258]: what if the check fails? Then the includeManuallyAddedLocalRoles will be undefined...

Comment (ChamathNS) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java:258]: booleans are by default false right?

**Comment by rksk** [comment]
Please note when backporting: This PR caused a regression https://github.com/wso2/product-is/issues/12081

**Pull request merged**